src/ofdm.c

```diff
@@ -1140,12 +1158,56 @@ static float est_timing_and_freq(struct OFDM *ofdm,
         complex float mvec[Npsam];
         for(int i=0; i<Npsam; i++) {
             complex float ph = cmplx(w*i);
-            mvec[i] = known_samples[i]*ph;
+            mvec[i] = conjf(known_samples[i]*ph);
         }
         for(int t=0; t<Ncorr; t+=tstep) {
+#if USE_VECTOR_OPS
+            float *rxPtr = (float*)&rx[t];
+            float *vecPtr = (float*)mvec;
+            float4 accumPos = { 0, 0, 0, 0 };
+            float4 accumNeg = { 0, 0, 0, 0 };
+            float4 accumImag = { 0, 0, 0, 0 };
+            float corrR = 0, corrI = 0;
+            int numBlocks = Npsam >> 1;
+            for (int i = 0; i < numBlocks; i++)
+            {
+                /* Lay out vectors as follows: 
+                   vec1 = rx[0].a, rx[0].b, rx[1].a, rx[1].b, ...
+                   vec2 = mvec[0].c, mvec[0].d, mvec[1].c, mvec1[1].d, ... */
+                float4 vec1 = { rxPtr[0], rxPtr[1], rxPtr[2], rxPtr[3] };
+                float4 vec2 = { vecPtr[0], vecPtr[1], vecPtr[2], vecPtr[3] };
+
```

Comment:
These assignments could be chewing up some CPU.  Could we use pointers `pvec1` & `pvec2` so the inner loop is just the vector operations?

Reply:
gcc seems to be using the following assembly for the assignments so I suspect there won't be much of a difference:

```
                float4 vec2 = { vecPtr[0], vecPtr[1], vecPtr[2], vecPtr[3] };
   26740:       f3 0f 10 52 0c          movss  0xc(%rdx),%xmm2
                float4 vec1 = { rxPtr[0], rxPtr[1], rxPtr[2], rxPtr[3] };
   26745:       f3 0f 10 28             movss  (%rax),%xmm5
```